pyproject.toml

```diff
@@ -3,21 +3,22 @@ requires = ["hatchling>=1.22.2", "hatch-vcs>=0.3.0"]
 build-backend = "hatchling.build"
 
 [project]
-name = "stac_geoparquet"
+name = "stac-geoparquet"
 authors = [{ name = "Tom Augspurger", email = "taugspurger@microsoft.com" }]
 readme = "README.md"
 license = { file = "LICENSE" }
 classifiers = ["License :: OSI Approved :: MIT License"]
 dynamic = ["version", "description"]
-requires-python = ">=3.8"
+requires-python = ">=3.10"
```

Comment:
Why the jump all the way to 3.10? 3.9 isn't EOL yet.

Reply:
There's 3.10 typing in this repo so 3.9 tests broke: https://github.com/stac-utils/stac-geoparquet/actions/runs/13201302550/job/36853751522

Reply:
I'd argue the better way to fix that is to put 
```py
from __future__ import annotations
```
at the top of that file but maintain 3.9 compatibility for now

Reply:
Does that allow `str | None` syntax?

Reply:
Agreed, this repo had already implicitly gone to 3.10 so I was just bringing the declaration in line. I'll go downgrade the syntax to keep 3.9.

Reply:
> Does that allow `str | None` syntax?

IIRC it does because the types don't get evaluated. We'd have to double check in CI though.

> implicitly gone to 3.10 

carelessly and accidentally... let's not look at that Git blame 🫣

Reply:
Yup, `__future__` works. Thanks. 🙇🏼